server/src/main/java/org/opensearch/index/snapshots/blobstore/RemoteStoreShardShallowCopySnapshot.java

```diff
@@ -0,0 +1,324 @@
+                }
+            } else {
+                parser.skipChildren();
+            }
+        }
+
+        return new RemoteStoreShardShallowCopySnapshot(
+            snapshot,
+            indexVersion,
+            primaryTerm,
+            commitGeneration,
+            startTime,
+            time,
+            totalFileCount,
+            totalSize,
+            indexUUID,
+            remoteStoreRepository,
+            repositoryBasePath,
+            fileNames
+        );
+    }
+
+    /**
+     * Returns shard primary Term during snapshot creation
+     *
+     * @return primary Term
+     */
+    public long getPrimaryTerm() {
+        return primaryTerm;
+    }
+
+    /**
+     * Returns snapshot commit generation
+     *
+     * @return commit Generation
+     */
+    public long getCommitGeneration() {
+        return commitGeneration;
+    }
+
+    /**
+     * Returns Index UUID
+     *
+     * @return index UUID
+     */
+    public String getIndexUUID() {
+        return indexUUID;
+    }
+
+    public String getRemoteStoreRepository() {
+        return remoteStoreRepository;
+    }
+
+    /**
+     * Returns snapshot name
+     *
+     * @return snapshot name
+     */
+    public String snapshot() {
+        return snapshot;
+    }
+
+    /**
+     * Returns list of files in the shard
+     *
+     * @return list of files
+     */
+
+    /**
+     * Returns snapshot start time
+     */
+    public long startTime() {
+        return startTime;
+    }
+
+    /**
+     * Returns snapshot running time
+     */
+    public long time() {
+        return time;
+    }
+
+    /**
+     * Returns incremental of files that were snapshotted
+     */
+    public int incrementalFileCount() {
+        return 0;
+    }
+
+    /**
+     * Returns total number of files that are referenced by this snapshot
+     */
+    public int totalFileCount() {
+        return totalFileCount;
+    }
+
+    /**
+     * Returns incremental of files size that were snapshotted
+     */
+    public long incrementalSize() {
+        return 0;
+    }
+
+    /**
+     * Returns total size of all files that where snapshotted
+     */
+    public long totalSize() {
+        return totalSize;
+    }
+
+    /**
+     * Creates a new instance which has a different name and zero incremental file counts but is identical to this instance in terms of the files
+     * it references.
+     *
+     * @param targetSnapshotName target snapshot name
+     * @param startTime          time the clone operation on the repository was started
+     * @param time               time it took to create the clone
+     */
+    public RemoteStoreShardShallowCopySnapshot asClone(String targetSnapshotName, long startTime, long time) {
+        return new RemoteStoreShardShallowCopySnapshot(
+            targetSnapshotName,
+            indexVersion,
+            primaryTerm,
+            commitGeneration,
+            startTime,
+            time,
+            0,
+            0,
```

Comment:
Oh, I initially had these 0s for incremental file count and size but since then we made a change as part of create API changes to have default value of 0 for those fields. Updated in new revision.